329 go native or go home

jimmykarily opened the pull request:
Fixes #329

jimmykarily commented:
Still missing the cleanup of the org/namespaces when doing `epinio uninstall`

jimmykarily commented:
By moving the staging to a different namespace, this PR broke the staging logs of `epinio push --verbosity 1`. This is because of the way we tail for logs here: https://github.com/epinio/epinio/blob/main/internal/cli/clients/client.go#L1347

(changed here: https://github.com/epinio/epinio/pull/337/files#diff-9175d7addc590890bbee03a3eaaf4fedbee6f21a2f87d5993d3b17296283382dR1468-R1469)

We need to find a way to match the staging pods on the tekton-staging namespace too.

manno reviewed (approved):
LGTM

(maybe squash some commits and write an abstract?)

manno commented on internal/cli/clients/client.go:
```diff
@@ -1274,15 +1287,124 @@ spec:
+	if err != nil {
+		return errors.Wrap(err, "failed to create file for application Service definition")
+	}
+	defer func() { err = serviceFile.Close() }()
```
While this does make the linter happy, it does not check the err. I prefer 
```suggestion
	defer func() { _ = serviceFile.Close() }()
```

Or we disable the errchk linter?

jimmykarily replied:
done

jimmykarily commented:
I squashed most commits on this one: https://github.com/epinio/epinio/pull/337/commits/bb2140351b7774d8499fcd5483c64c0f366993e2 and added an abstract in the commit message

jimmykarily commented:
I have a fix for the logs on another branch. I will open a PR as soon as we merge this one.

jimmykarily commented:
PR that fixes logs: https://github.com/epinio/epinio/pull/339
